CRM/Utils/System/Drupal8.php

```diff
@@ -64,19 +65,29 @@ public function createUser(&$params, $mail) {
     if ($user_register_conf != 'visitors' && !$user->hasPermission('administer users')) {
       $account->block();
     }
+    elseif (!$verify_mail_conf) {
+      $account->activate();
+    }
 
     // Validate the user object
     $violations = $account->validate();
     if (count($violations)) {
       return FALSE;
     }
 
+    // Let the Drupal module know we're already in CiviCRM.
+    $config = CRM_Core_Config::singleton();
+    $config->inCiviCRM = TRUE;
```

Comment:
Just a note that setting this variable and using it in this way is taken from the Drupal 7 code - I didn't just make this up :-)